ballerina/client.bal

```diff
@@ -341,6 +341,192 @@ public isolated client class Client {
     }
+
+    # Initiates a multipart upload and returns an upload ID.
+    #
+    # + objectName - The name of the object  
+    # + bucketName - The name of the bucket  
+    # + cannedACL - The access control list of the new object
+    # + multipartUploadHeaders - Optional headers for multipart uploads
+    # + return - If success, the upload ID, else an error
+    remote isolated function createMultipartUpload(
+            @display {label: "Object Name"} string objectName,
+            @display {label: "Bucket Name"} string bucketName,
+            @display {label: "Grant"} CannedACL? cannedACL = (),
+            @display {label: "Multipart Upload Headers"} MultipartUploadHeaders? multipartUploadHeaders = ())
+            returns string|error {
+
+        if objectName == EMPTY_STRING {
+            return error(EMPTY_OBJECT_NAME_ERROR_MSG);
+        }
+        if bucketName == EMPTY_STRING {
+            return error(EMPTY_BUCKET_NAME_ERROR_MSG);
+        }
+
+        http:Request request = new;
+        
+        string requestURI = string `/${bucketName}/${objectName}`;
+        string queryParamStr = string `?uploads`;
+        map<string> requestHeaders = setDefaultHeaders(self.amazonHost);
+        if cannedACL is CannedACL {
+            requestHeaders[X_AMZ_ACL] = cannedACL;
+        }
+        populateMultipartUploadHeaders(requestHeaders, multipartUploadHeaders);
+
+        check generateSignature(self.accessKeyId, self.secretAccessKey, self.region, POST, requestURI, UNSIGNED_PAYLOAD,
+            requestHeaders, request, queryParams = {"uploads": EMPTY_STRING});
+        requestURI = string `${requestURI}${queryParamStr}`;
+        http:Response httpResponse = check self.amazonS3->post(requestURI, request);
+
+        xml XMLPayload = check httpResponse.getXmlPayload();
+        if httpResponse.statusCode == http:STATUS_OK {
+            return getUploadId(XMLPayload);
+        } else {
+            return error(XMLPayload.toString());
+        }
+    }
+    
+    # Completes a multipart upload by assembling previously uploaded parts.
+    #
+    # + objectName - The name of the object  
+    # + bucketName - The name of the bucket  
+    # + payload - The file content that needed to be added to the bucket  
+    # + uploadId - The upload ID of the multipart upload  
+    # + partNumber - The part number of the object  
+    # + uploadPartHeaders - Optional headers for the upload
+    # + return - An error on failure or else `()`
+    remote isolated function uploadPart(
+            @display {label: "Object Name"} string objectName,
+            @display {label: "Bucket Name"} string bucketName,
+            @display {label: "File Content"} string|xml|json|byte[]|stream<io:Block, io:Error?> payload,
+            @display {label: "Upload ID"} string uploadId,
+            @display {label: "Part Number"} int partNumber,
+            @display {label: "UploadPartHeaders"} UploadPartHeaders? uploadPartHeaders = ())
```

Comment:
```suggestion
            @display {label: "UploadPartHeaders"} UploadPart Headers? uploadPartHeaders = ())
```

Reply:
UploadPartHeaders is the name of record

Reply:
UploadPartHeaders is the name of the record

Reply:
@nisuraaa my bad :) I've mistakenly edited the type name instead of the label. Edited the suggestion